packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneAndTalkButton/DoneAndTalkButton.spec.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import { expect } from 'chai'
+import sinon from 'sinon'
+import DoneAndTalkButton, { StyledDoneAndTalkButton } from './DoneAndTalkButton'
+
+describe('DoneAndTalkButton', function () {
+  it('should render without crashing', function () {
+    const wrapper = mount(<DoneAndTalkButton />)
+    expect(wrapper).to.be.ok
+  })
+  
+  describe('when props.completed is true', function () {
+    it('should render null', function () {
+      const wrapper = mount(<DoneAndTalkButton completed />)
+      expect(wrapper.html()).to.be.null
+    })
+  })
+
+  it('should call props.onClick for the onClick event', function () {
+    const onClickSpy = sinon.spy()
+    const wrapper = mount(<DoneAndTalkButton onClick={onClickSpy} />)
+    wrapper.find('button').simulate('click')
+    expect(onClickSpy.calledOnce).to.be.true
+  })
+
+  describe('when props.completed is false', function () {
+    it('should render a ThemeProvider', function () {
+      const wrapper = mount(<DoneAndTalkButton />)
+      expect(wrapper.find('ThemeProvider')).to.have.lengthOf(1)
+    })
+
+    it('should render a StyledDoneAndTalkButton', function () {
+      const wrapper = mount(<DoneAndTalkButton />)
+      expect(wrapper.find(StyledDoneAndTalkButton)).to.have.lengthOf(1)
+    })
+  })
+})
```

Comment:
The component instance has a lot of props but only a couple are tested here. Should there be tests for the other props eg. project slug or subject ID?

Reply:
I'm hold off on testing some of them because they aren't actually functionally implemented in the rewrite yet like the gold standard mode. I know the styles for some of these things have changed, so I'm not sure if this button really needs some of these props still.